Toasts on failed login & password visibility toggle & no server field auto correct

ThibaultNocchi opened the pull request:
I added two toasts: one when the server can't be reached (it also triggers when no login or password are filled but the server exists, should this edge case be fixed?) and one when login credentials are rejected by the server.

Fixes #13

dkanada commented on app/src/main/java/com/dkanada/gramophone/ui/activities/LoginActivity.java:
```diff
@@ -119,6 +120,10 @@ public void onResponse(AuthenticationResult result) {
                             if (result.getAccessToken() == null) return;
                             check(context, serverCredentials, result);
                         }
+                        @Override
```
```suggestion

                        @Override
```

dkanada commented:
> Should this edge case be fixed?

Ideally yes, but I'll leave that up to you! I also really need to fix the URL input so it doesn't think domains are a wrong word and add a "show password" icon, but I haven't had the time.

ThibaultNocchi commented:
Ok so I've added a "missing username" toast if there is no username in the field. Do you agree that to connect to a Jellyfin server you'll always need a non empty username?

Also, I've added a visibility toggle to the password field and disabled auto correction to the server field (that's what you were saying by "so it doesn't think domains are a wrong word"?)

Finally, I'd advocate using the squash and merge feature when closing a PR, especially with PR such as mines with many "noise" commits, what do you think?

dkanada commented:
I agree you'll always need a non-empty username, and I was indeed talking about autocorrect haha, was a bit tired and couldn't think of the word. Squash merge sounds like a good idea but GitHub recently changed it so the person merging gets credit for squashes (such a stupid idea) so if you could squash it and force push that would be best.

ThibaultNocchi commented:
Well great so I've added what you wanted, nice! :)

I didn't know that. It's a first for me the force push, but it seems to not bother the PR. Thanks!

dkanada commented on app/src/main/java/com/dkanada/gramophone/ui/activities/LoginActivity.java:
```diff
@@ -107,6 +107,11 @@ public void onClick(View v) {
                 return;
             }
 
+            if(username.getText().toString().trim().length()==0){
```
```suggestion
            if (username.getText().toString().trim().length()==0) {
```

dkanada commented on app/src/main/java/com/dkanada/gramophone/ui/activities/LoginActivity.java:
```diff
@@ -107,6 +107,11 @@ public void onClick(View v) {
                 return;
             }
 
+            if (username.getText().toString().trim().length()==0) {
```
```suggestion
            if (username.getText().toString().trim().length() == 0) {
```
